### ISSUE TITLE
Failure to deploy kubernetes-anywhere

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-tls#8]
_This issue was originally opened by @dtonnesen as hashicorp/terraform#16171. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.

Terraform v0.9.4 (built with kubernetes-anywhere don't know if upgrade is possible)

### Terraform Configuration Files

# Copy-paste your Terraform configurations here.
#

As this is kubernetes-anywhere I am not sure what config files are required.

# For large Terraform configs, please use a service like Dropbox and
# share a link to the ZIP file. For security, you can also encrypt the
# files using our GPG public key.
```

### Debug Output
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.

### Crash Output
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.

https://gist.github.com/dtonnesen/23303cae480b917f3b7344d1b5cecb38#file-crash-log

### Expected Behavior
What should have happened?

Complete successfully

### Actual Behavior
What actually happened?

Crashed

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:

1. Run **make deploy**

### Important Factoids
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?

Yes it runs in kubernetes-anywhere

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

**Comment by SpencerBrown** [comment]
Checking back to the [original issue](https://github.com/hashicorp/terraform/issues/16171), the line of code causing the panic is `resource_cert_request.go:85`. This panic was fixed by #7, and should now produce the error message `subject block cannot be empty`. The root cause of this crash is a `tls_cert_request` resource with an empty or missing `subject` block.

Retrying this test with release 1.0.1 of the TLS provider should produce the error message instead of a crash. Please try it and let us know.

**Comment by SpencerBrown** [comment]
Hi @apparentlymart, wondering how you want to handle this issue? See my investigation in my November comment above, it seems pretty certain that the issue is resolved, and @dtonnesen isn't responding. Is it reasonable to close this, or what would the normal process be?

**Comment by apparentlymart** [comment]
Hi @SpencerBrown! Thanks for checking in here.

I agree with your analysis... this looks like the same bug that was fixed in #7, and so I'm going to close this. @dtonnesen if you are still seeing this problem after upgrading to the latest version then please come back and open a _new_ issue in this repository (_not_ the Terraform Core repository), just because it's easy to miss comments in closed issues.
